src/odemis/gui/cont/features.py

```diff
@@ -0,0 +1,155 @@
+import logging
+
+import wx
+
+from odemis.acq.feature import FEATURE_ACTIVE, FEATURE_MILLED, FEATURE_DEACTIVE
+from odemis.gui.model import TOOL_FEATURE
+from odemis.gui.util.widgets import VigilantAttributeConnector
+
+
+class CryoFeatureController(object):
+    """ controller to handle the cryo feature panel elements
+    It requires features list VA & currentFeature VA on the tab data to function properly
+    """
+
+    def __init__(self, tab_data, panel, tab):
+        """
+        tab_data (MicroscopyGUIData): the representation of the microscope GUI
+        panel (wx._windows.Panel): the panel containing the UI controls
+        tab: (Tab): the tab which should show the data
+        """
+        if not hasattr(tab_data, 'features'):
+            raise ValueError("features list VA is required.")
+
+        if not hasattr(tab_data, 'currentFeature'):
+            raise ValueError("currentFeature VA is required.")
+
+        self._tab_data_model = tab_data
+        self._main_data_model = tab_data.main
+        self._panel = panel
+        self._tab = tab
+
+        self._tab_data_model.features.subscribe(self._on_features_changes, init=True)
+        self._tab_data_model.currentFeature.subscribe(self._on_current_feature_changes, init=True)
+
+        # features va attributes (name, status..etc) connectors
+        self._feature_name_va_connector = None
+        self._feature_status_va_connector = None
+        self._feature_z_va_connector = None
+
+        # Event binding
+        self._panel.cmb_features.Bind(wx.EVT_COMBOBOX, self._on_cmb_features_change)
+        self._panel.btn_create_move_feature.Bind(wx.EVT_BUTTON, self._on_btn_create_move_feature)
+        self._panel.btn_go_to_feature.Bind(wx.EVT_BUTTON, self._on_btn_go_to_feature)
+        self._panel.btn_use_current_z.Bind(wx.EVT_BUTTON, self._on_btn_use_current_z)
+
+    def _on_btn_create_move_feature(self, _):
+        # As this button is identical to clicking the feature tool,
+        # directly change the tool to feature tool
+        self._tab_data_model.tool.value = TOOL_FEATURE
+
+    def _on_btn_use_current_z(self, _):
+        # Use current focus to set currently selected feature
+        feature = self._tab_data_model.currentFeature.value
+        if feature:
+            pos = feature.pos.value
+            current_focus = self._main_data_model.focus.position.value['z']
+            feature.pos.value = (pos[0], pos[1], current_focus)
+
+    def _on_btn_go_to_feature(self, _):
+        """
+        Move the stage and focus to the currently selected feature
+        """
+        feature = self._tab_data_model.currentFeature.value
+        if not feature:
+            return
+        pos = feature.pos.value
+        logging.info(f"Moving to position: {pos}")
+        self._main_data_model.stage.moveAbs({'x': pos[0], 'y': pos[1]})
+        self._main_data_model.focus.moveAbs({'z': pos[2]})
+
+    def _on_features_changes(self, features):
```

Comment:
This function makes changes on the GUI, so you could add the decorator `call_in_wx_main`

Reply:
The `call_in_wx_main` decorator is called from inside the VA connector. Check the implementation of `VigilantAttributeConnector` class.